state/cassandra/cassandra.go

```diff
@@ -0,0 +1,304 @@
+	if err != nil {
+		return nil, err
+	}
+
+	cluster.Consistency = cons
+	return cluster, nil
+}
+
+func (c *Cassandra) getConsistency(consistency string) (gocql.Consistency, error) {
+	switch consistency {
+	case "All":
+		return gocql.All, nil
+	case "One":
+		return gocql.One, nil
+	case "Two":
+		return gocql.Two, nil
+	case "Three":
+		return gocql.Three, nil
+	case "Quorum":
+		return gocql.Quorum, nil
+	case "LocalQuorum":
+		return gocql.LocalQuorum, nil
+	case "EachQuorum":
+		return gocql.EachQuorum, nil
+	case "LocalOne":
+		return gocql.LocalOne, nil
+	case "Any":
+		return gocql.Any, nil
+	case "":
+		return defaultConsistency, nil
+	}
+	return 0, fmt.Errorf("consistency mode %s not found", consistency)
+}
+
+func getCassandraMetadata(metadata state.Metadata) (*cassandraMetadata, error) {
+	meta := cassandraMetadata{
+		protoVersion:      defaultProtoVersion,
+		table:             defaultTable,
+		keyspace:          defaultKeyspace,
+		replicationFactor: defaultReplicationFactor,
+		consistency:       "All",
+	}
+
+	if val, ok := metadata.Properties[hosts]; ok && val != "" {
+		meta.hosts = strings.Split(val, ",")
+	} else {
+		return nil, errors.New("missing or empty hosts field from metadata")
+	}
+
+	if val, ok := metadata.Properties[consistency]; ok && val != "" {
+		meta.consistency = val
+	}
+
+	if val, ok := metadata.Properties[table]; ok && val != "" {
+		meta.table = val
+	}
+
+	if val, ok := metadata.Properties[keyspace]; ok && val != "" {
+		meta.keyspace = val
+	}
+
+	if val, ok := metadata.Properties[username]; ok && val != "" {
+		meta.username = val
+	}
+
+	if val, ok := metadata.Properties[password]; ok && val != "" {
+		meta.password = val
+	}
+
+	if val, ok := metadata.Properties[protoVersion]; ok && val != "" {
+		p, err := strconv.ParseInt(val, 0, 32)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing protoVersion field: %s", err)
+		}
+		meta.protoVersion = int(p)
+	}
+
+	if val, ok := metadata.Properties[replicationFactor]; ok && val != "" {
+		r, err := strconv.ParseInt(val, 0, 32)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing replicationFactor field: %s", err)
+		}
+		meta.replicationFactor = int(r)
+	}
+
+	return &meta, nil
+}
+
+// Delete performs a delete operation
+func (c *Cassandra) Delete(req *state.DeleteRequest) error {
+	query := fmt.Sprintf("DELETE FROM %s WHERE key='%s'", c.table, req.Key)
+	err := c.session.Query(query).Exec()
+	return err
+}
+
+// BulkDelete performs a bulk delete operation
+func (c *Cassandra) BulkDelete(req []state.DeleteRequest) error {
+	for _, re := range req {
+		err := c.Delete(&re)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Get retrieves state from cassandra with a key
+func (c *Cassandra) Get(req *state.GetRequest) (*state.GetResponse, error) {
+	query := fmt.Sprintf("SELECT value FROM %s WHERE key='%s'", c.table, req.Key)
+
+	session := c.session
+
+	if req.Options.Consistency == state.Strong {
```

Comment:
Isn't there any chance that user passes unsupported consistency ? e.g. typo. evntual

Reply:
Yes. And then the default consistency level is used